pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -138,6 +140,19 @@ func validateStepOverrides(overrides []TaskRunStepOverride) (errs *apis.FieldErr
 	return errs
 }
 
+// validateTaskRunComputeResources ensures that compute resources are not configured at both the step level and the task level
+func validateTaskRunComputeResources(computeResources *corev1.ResourceRequirements, overrides []TaskRunStepOverride) (errs *apis.FieldError) {
+	for _, override := range overrides {
+		if override.Resources.Size() > 0 && computeResources != nil {
```

Comment:
can you use `if len(overrides.Resources) !=0 ...`?

Reply:
hmm seems a bit better. != is a bit more specific condition for >